src/app/components/Messages/UserMessage.vue

```diff
@@ -53,40 +86,221 @@
+            });
+          } else {
+            console.log(`No such user ${user}`);
+          }
+        });
+        // Hide the tooltip not hovered over
+        l.addEventListener('mouseleave', (ev) => {
+          this.tooltipVisible = false;
+        });
+        // When the mouse is moved, update the tooltip position
+        l.addEventListener('mousemove', (e: Event) => {
+          const ev = e as MouseEvent;
+          this.tooltipX = ev.clientX;
+          this.tooltipY = ev.clientY;
+        });
+        // Opens player profile on click
+        l.addEventListener('click', () => {
+          const username = l.innerHTML.substring(1);
+          const user = this.$vxm.chat.connectedUsers[username];
+          if (!user) return;
+          window.open(`https://${this.$vxm.chat.workbranch}/web/player/${user.uid}/`);
+        });
+      });
+      // Gets puzzle links from markdown
+      const puzzleLinks = this.$refs.msg.getElementsByClassName('puzzle-link');
+      // Spread operators converts HTMLCollectionOf<Element> to HTMLElement[]
+      // Iterates through each channel link
+      [...puzzleLinks].forEach(l => {
+        // Adds event listener for hover
+        l.addEventListener('mouseenter', (ev) => {
+          const url = (l as HTMLElement).innerText;
+          const matches = url.match(/\d+\/?$/);
+          this.pid = parseInt((matches ? matches[0] : '0'), 10);
+          this.puzzleTooltipVisible = true; // Make it visible
+            /* Ensures information is loaded after the tooltip exists
+            The tooltip uses v-if (to limit GET requests)
+            If the functions are called right away,
+            they could fail because the tooltip doesn't exist yet */
+            this.$nextTick(() => {
+              // Updates tooltip information
+              this.$refs.puzzleTooltip.fill();
+          });
+        });
+        // Hide the tooltip not hovered over
+        l.addEventListener('mouseleave', (ev) => {
+          this.puzzleTooltipVisible = false;
+        });
+        // When the mouse is moved, update the tooltip position
+        l.addEventListener('mousemove', (e: Event) => {
+          const ev = e as MouseEvent;
+          this.puzzleTooltipX = ev.clientX;
+          this.puzzleTooltipY = ev.clientY;
+        });
+      });
+      const usernames = Object.keys(this.$vxm.chat.connectedUsers);
+      [...this.$el.getElementsByTagName('a')]
+        .filter(e => e.href !== e.innerText) // If the text doesn't match the href
+        .filter(e => !usernames.includes(e.innerText.replace('●', '').trim().replace(':', '').trim())) // Don't flag usernames
+        .filter(e => ![...e.classList].includes('chat-message-options')) // Or context menu
+        .filter(e => ![...e.classList].includes('user-link')) // Or user tooltips
+        .filter(e => ![...e.classList].includes('image-link')) // Or inline images
+        .forEach(e => {
+        e.addEventListener('click', ev => {
```

Comment:
Potentially yeah. You’d need to keep around references you can use to destroy the listeners.

As I’m thinking of it though, shouldn’t it be possible to tie into the markdown library to do this? I’m pretty sure it should be possible to register a handler for any links that get added in

Reply:
I've moved this to markdown, which has simplified the relevant code. However, given that I can't test message editing at the moment, I think it's best to hold off for now - once message editing is enabled, this will need to be done anyway.